src/app/pages/dashboard/components/widget-group-form/widget-group-form.component.ts

```diff
@@ -1,47 +1,161 @@
-    template: [''],
-    layout: [WidgetGroupLayout.Full],
-  });
-  readonly layoutsMap = widgetGroupIcons;
+
+  protected readonly template = new FormControl('');
+  protected readonly layout = new FormControl(WidgetGroupLayout.Full, [Validators.required]);
+  protected readonly category = new FormControl(null as WidgetCategory, [Validators.required]);
+  protected readonly type = new FormControl(null as WidgetType, [Validators.required]);
+  protected readonly layoutsMap = widgetGroupIcons;
+  protected readonly widgetRegistry = widgetRegistry;
+
+  widgetCategoriesOptions$: Observable<Option[]>;
+  widgetTypesOptions$: Observable<Option[]>;
 
   // TODO: Implement template options
   templateOptions$ = of([]);
 
   constructor(
-    private formBuilder: FormBuilder,
-    private chainedRef: ChainedRef<WidgetGroup>,
+    protected chainedRef: ChainedRef<WidgetGroup>,
     private cdr: ChangeDetectorRef,
+    protected widgetGroupFormStore: WidgetGroupFormStore,
   ) {
-    this.group = chainedRef.getData();
-    this.form.controls.layout.setValue(this.group.layout);
-    this.form.controls.layout.valueChanges
-      .pipe(untilDestroyed(this))
-      .subscribe((layout) => {
+    this.setCategoryOptions();
+    this.setInitialFormValues();
+    this.setupLayoutUpdates();
+    this.setupCategoryUpdate();
+    this.setupTypeUpdate();
+  }
+
+  setupTypeUpdate(): void {
+    this.type.valueChanges.pipe(
+      filter(Boolean),
+      untilDestroyed(this),
+    ).subscribe({
+      next: (type) => {
+        this.widgetGroupFormStore.setType({ slotIndex: this.selectedSlot, type });
+      },
+    });
+  }
+
+  setInitialFormValues(): void {
+    this.group = this.chainedRef.getData();
+    this.layout.setValue(this.group.layout);
+  }
+
+  setupLayoutUpdates(): void {
+    this.layout.valueChanges.pipe(
+      tap((layout) => {
+        this.widgetGroupFormStore.setLayout(layout);
         this.group = { ...this.group, layout };
+        this.resetSlot();
         this.cdr.markForCheck();
-      });
+      }),
+      untilDestroyed(this),
+    ).subscribe();
+  }
+
+  setupCategoryUpdate(): void {
+    this.category.valueChanges.pipe(
+      filter(Boolean),
+      tap((category) => {
+        this.widgetGroupFormStore.setCategory({ slotIndex: this.selectedSlot, category });
+        this.setWidgetTypeOptions(category);
+      }),
+      untilDestroyed(this),
+    ).subscribe();
+  }
+
+  selectedSlotChanged(slotIndex: number): void {
+    this.selectedSlot = slotIndex;
+    this.resetSlot();
+  }
+
+  resetSlot(): void {
+    this.category.setValue(null);
+    this.type.setValue(null);
+    this.setCategoryOptions();
+  }
+
+  getEnumKeyByEnumValue<T extends Record<string, string>>(
+    myEnum: T,
+    enumValue: string,
+  ): keyof T | null {
+    const keys = Object.keys(myEnum).filter((x) => myEnum[x] === enumValue);
+    return keys.length > 0 ? keys[0] : null;
+  }
+
+  setCategoryOptions(): void {
```

Comment:
This is complicated and you are doing similar things here and in `setWidgetTypeOptions`.
Instead, why not first filter for widgets that fit the slot size in a private method:

```ts
    const widgets = Object.values(widgetRegistry);
    const layout = this.layout.value;
    const slotSize = layoutToSlotSizes[layout][this.selectedSlot];

    const supportedWidgets = widgets.filter((widget) => widget.supportedSizes.includes(slotSize));
```

and then extract categories:
```ts
    const categories = uniq(supportedWidgets.map((widget) => widget.category));
    return categories.map((category) => {
      return {
        label: widgetCategoryLabels.get(category) || category,
        value: category
      };
    });
```

or types when you need them.